spring-data-relational/src/main/java/org/springframework/data/relational/core/sql/CaseExpression.java

```diff
@@ -0,0 +1,94 @@
+package org.springframework.data.relational.core.sql;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static java.util.stream.Collectors.joining;
+
+/**
+ * Case with one or more conditions expression.
+ * <p>
+ * Results in a rendered condition:
+ * <pre>
+ *   CASE
+ *     WHEN condition1 THEN result1
+ *     WHEN condition2 THEN result2
+ *     ELSE result
+ *   END
+ * </pre>
+ * </p>
+ *
+ * @author Sven Rienstra
+ * @since 3.4
+ */
+public class CaseExpression extends AbstractSegment implements Expression {
+    private final List<When> whenList = new ArrayList<>();
+    private final Literal other;
```

Comment:
`other` is a undescriptive as it gets. I guess we can't use `else` or `default` since those are Java keywords. 

But `elseBranch` would be a better name, or maybe `whenElse`.

Reply:
Also `other` should be an `Expression`, not constraint to a `Literal`